### DISCUSSION TITLE
Make Composer browsing behave like Artist browsing

### DISCUSSION THREAD

**Discussion by kreativkod** [rern/rAudio#963] [Feature Requests]
@rern 
When clicking a composer in the Composers list, I would argue it is, not exclusively, but very very common to have an idea in mind of the work of that composer you want to listen to (a symphony, sonata, trio, etc). However, the list of available albums are sorted on artists. 

People have different tastes and needs, lets say it split even 50/50 between those who want to have the list sorted on ID3 artist and those who prefer having the list sorted according to ID3 album. One preference is not better than the other. 

But, and this is a big but, the current sorting on ID3 artist collapses when we realise that an album may be tagged with several artists. Since rAudio will only list the album once in the list one does not know under what artist it will show up. One has to remember, scroll to find or  have other extra strategies to find the thing one wants to listen to.

I argue a more consistent and predictable approach would be to reverse the list sorting, e.g. ID3 album - ID3 artist, rather than the other way around, so that when I click e.g. "Gustav Mahler" all "Symphony no. 1" and "Symphony no. 3" and "Des Knaben Wunderhorn" would show up next to one another rather than being spread out across the list depending on which artist is chosen by the script, given that there even is an artist which is not necessarily the case when it comes to e.g. symphonies.

1) Could you implement an option similar to "Sort Album by Artist" that sorts the resulting list when clicking a composer on the ID3 album tag rather than on the artist tag?

~~2) Or point me to a place in the code where I could change the default `$filemode` to `album` rather than `byartist`? (I think that is what is needed). I don't mind if this is reset by updates, I can redo the fix.~~

**Comment by kreativkod** [comment]
I found a way to switch the positions of artist and album in the code. For anyone who is interested:

file: `/srv/http/mpdlibrary.php`
On line 354: `$name = $fL > 1 && $mode !== 'conductor' ? $val1.'<gr> • </gr>'.$val0 : $val0;`
Switch positions of `$val1` and the first `$val0`. That takes care of the presentation, but the list is still sorted on artist.
So, on line 323: `$sort = in_array( $mode, [ 'artist', 'albumartist' ] ) ? $list[ 0 ] : $list[ 1 ]; // sort by artist`
Switch 0 for 1 and 1 for 0.

This also affects album display in e.g., genre and date views. The presentational aspect could probably easily be constrained to composer mode only by duplicating and amending the `if` test, but the sorting would still apply to all modes and thus need a solution that is not as obvious to me.

**Comment by rern** [comment]
Your argument is valid.
`Composer` will list `album • artist` by next update.
